Subject: [PATCH] ipt_ECN corrupt checksum fix
Date:	Sun, 02 Jan 2005 17:40:33 +0000
X-BK-ChangeSetKey: rusty@rustcorp.com.au[torvalds]|ChangeSet|20050102174033|49608
Patch-mainline: yes
From: olh@suse.de


ChangeSet 1.2106, 2005/01/02 09:40:33-08:00, rusty@rustcorp.com.au

	[PATCH] ipt_ECN corrupt checksum fix
	
	Nasty bug, caught while writing the ECN target test.  Corrupts
	checksums of packets when target is used on them.
	
	Let this be a warning on the evils of casts.
	
	Signed-off-by: Linus Torvalds <torvalds@osdl.org>



 ipt_ECN.c |    2 +-
 1 files changed, 1 insertion(+), 1 deletion(-)


diff -Nru a/net/ipv4/netfilter/ipt_ECN.c b/net/ipv4/netfilter/ipt_ECN.c
--- a/net/ipv4/netfilter/ipt_ECN.c	2005-01-02 10:08:54 -08:00
+++ b/net/ipv4/netfilter/ipt_ECN.c	2005-01-02 10:08:54 -08:00
@@ -67,7 +67,7 @@
 
 	if (einfo->operation & IPT_ECN_OP_SET_CWR)
 		th->cwr = einfo->proto.tcp.cwr;
-	diffs[1] = ((u_int16_t *)&th)[6];
+	diffs[1] = ((u_int16_t *)th)[6];
 
 	/* Only mangle if it's changed. */
 	if (diffs[0] != diffs[1]) {
-
To unsubscribe from this list: send the line "unsubscribe bk-commits-head" in
the body of a message to majordomo@vger.kernel.org
More majordomo info at  http://vger.kernel.org/majordomo-info.html

